frontend/src/entities/dataset/dataset.actions.tsx

```diff
@@ -161,70 +242,14 @@ export const DatasetBrowserActions = (state: Pick<DatasetBrowserState, 'selected
-                        const filesToUpload = Array.from(event.target?.files || []).map((file: File): DatasetResourceObject => ({
-                            bucket: '',
-                            type: 'object',
-                            key: `${state.datasetContext?.location || ''}${file.name}`,
-                            size: file.size,
-                            file,
-                            name: file.name,
-                        }));
-
-                        switch (manageMode) {
-                            case DatasetBrowserManageMode.Create:
-                                setState({
-                                    type: DatasetActionType.State,
-                                    payload: {
-                                        items: [...state.items, ...filesToUpload]
-                                    }
-                                });
-                                break;
-                            case DatasetBrowserManageMode.Edit:
-                                if (state.datasetContext) {
-                                    const filteringTextPrefix = prefixForPath(state.filteringText);
-                                    if (filteringTextPrefix) {
-                                        filesToUpload.forEach((file) => {
-                                            file.key = [filteringTextPrefix, file.key].join('');
-                                        });
-                                    }
-                                    
-                                    setDisableUpload(true);
-                                    // ensure cast to DatasetContext is valid
-                                    if (state.datasetContext.name && state.datasetContext.type) {
-                                        await uploadResources(state.datasetContext as DatasetContext, filesToUpload, notificationService);
-                                    }
-                                    setDisableUpload(false);
-
-                                    // if the filter contains a prefix append that to the Location
-                                    const effectiveContext = {...state.datasetContext, location: [state.datasetContext.location, filteringTextPrefix].join('')};
-                                    updateDatasetContext(effectiveContext, '', false);
-                                }
-                                break;
-                        }
-                    }}
-                    onBlur={() => {
-                        //This remains empty to regain keyboard focus
-                        //to the button after user closes the input window
-                        //for accessibility purposes
-                    }}
-                    multiple
-                />
+                <SpaceBetween size='xs' direction='horizontal'>
+                    <UploadButton state={state} setState={setState} updateDatasetContext={updateDatasetContext} isFileUpload={false}>
```

Comment:
Do we really need separate buttons for this? Generally when I upload files there aren't separate buttons. Can we not tell from the selected file context whether this is a folder or not?

Reply:
if we use pure html, then yes, we need to have separate buttons. We can potentially make it just a single button if I use some kind of library